packages/liveblocks-core/src/__tests__/Poller.test.ts

```diff
@@ -0,0 +1,146 @@
+import { makePoller } from "../lib/Poller";
+
+// Mock timers using Jest
+jest.useFakeTimers();
+
+describe("Poller", () => {
+  beforeEach(() => {
+    jest.useFakeTimers();
+  });
+
+  afterEach(() => {
+    jest.clearAllTimers();
+  });
+
+  test("should start the poller when condition is true", async () => {
+    const callback = jest.fn();
+    const poller = makePoller(callback, 1000); // 1000ms interval
+
+    poller.enable(true);
+
+    // Fast-forward until all timers have been executed
+    await jest.advanceTimersByTimeAsync(1000);
+
+    // Expect the callback to be called after the interval
+    expect(callback).toHaveBeenCalledTimes(1);
```

Comment:
Setting up the poller is great, but should we ideally structure all tests in this way?

```tsx
const poller = makePoller(callback, 1000);
try {
  // Run all tests
} finally {
  poller.enable(false);
}
```

Just to ensure that polling in the background actually stops as well?

I really wish we could use `using` here already!

```
using poller = makePoller(callback, 1000);
// Just run all tests 💆‍♂️
```